docs/roadmap.md

```diff
@@ -0,0 +1,91 @@
+# Roadmap
+
+This is a list of goals we have. Many of them are quite independent.
```

Comment:
(Optional:) Perhaps say

"This document lists some of the goals we have"

to make it clear this list is incomplete?

Reply:
Makes sense. Done.

Reply:
I was thinking also of how to communicate more of the sentiment from https://github.com/martinvonz/jj/pull/4332#discussion_r1730065786 about the intention of this roadmap, but this is the best I could come up so far.

Adding such a section could be good for a later update if you feel up to it.